pkg/sources/docker/docker.go

```diff
@@ -77,6 +77,35 @@ func (s *Source) Init(_ context.Context, name string, jobId sources.JobID, sourc
 		return fmt.Errorf("error unmarshalling connection: %w", err)
 	}
 
+	if namespace := s.conn.GetNamespace(); namespace != "" {
+		var registry Registry
+		switch {
+		case strings.HasPrefix(namespace, "quay.io"):
+			registry = &Quay{
+				Token: s.conn.GetRegistryToken(),
+			}
+		case strings.HasPrefix(namespace, "ghcr.io"):
+			registry = &GHCR{
+				Token: s.conn.GetRegistryToken(),
+			}
+		default: // default is dockerhub
+			registry = &DockerHub{
+				Token: s.conn.GetRegistryToken(),
+			}
+		}
```

Comment:
To improve readibility we can move this logic to a factory method like `MakeRegistryFromNamespace`